### PR TITLE
fix(button): prevent style changes on hover for disabled form buttons

### LINKED ISSUE #2270: button: shows border on hover when disabled
### I verified there's no existing issue for this bug.

- [X] There are no existing issues

### Current behavior

Buttons with `btn--form` modifier, on hover changes border color during disabled state. 

e.g, Listbox button with form modifier 
image
https://opensource.ebay.com/skin/storybook/?path=/story/skin-listbox-button-form--disabled

### Expected behavior

Expected as is

image

### Skin version

v17

### Steps to reproduce

https://opensource.ebay.com/skin/storybook/?path=/story/skin-listbox-button-form--disabled

### PR DESCRIPTION
**Pull request by saiponnada**
Fixes #2270

- [x] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
Fixed listbox button on hover during disabled state not to change the border color.

## Screenshots
**Before**
image

**After**
image

## Checklist

- [x] I verify the build is in a non-broken state
- [x] I verify all changes are within scope of the linked issue

- [x] I regenerated all CSS files under dist folder
- [x] I tested the UI in all supported browsers
- [x] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [x] I tested the UI in dark mode and RTL mode
- [ ] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

File: src/less/button/button.less [hunk 1]
```diff
@@ -354,7 +354,10 @@ a.fake-btn--form {
     border-color: inherit;
     .border-radius-token(expand-btn-border-radius, border-radius-50);
     max-width: 100%;
+}
 
+button.btn--form:not([disabled], [aria-disabled="true"]),
```
Comment (ArtBlue) [inline src/less/button/button.less:359]: The changes here are not in scope with the issue/PR titles. You seem to be applying the fix to listbox button as well as all buttons and fake buttons. Maybe update the issue and PR title.

Comment (ArtBlue) [inline src/less/button/button.less:359]: If we are including buttons in this change as well, why is this only applied to `btn--form` modified buttons? I'm not sure button outside of forms should be treated differently.

**Review by ArtBlue — COMMENTED** [review]
Some questions...

File: src/less/button/button.less (hunk 1 above)
Comment (saiponnada) [reply src/less/button/button.less:359]: Agree that buttons should not be treated any difference. But the way the code is structured today, disabled styles are applied at modifier level. For e.g.,

```
button.btn--primary.btn--destructive[disabled],
button.btn--primary.btn--destructive[aria-disabled="true"] {}

button.btn--primary.btn--destructive:not([disabled], [aria-disabled="true"]),
a.fake-btn--primary.fake-btn--destructive[href] {}
```

When we added `btn--form` with overrides, its overriding above styles with higher specificity. So, I added disabled styles for form modifiers as well to keep it consistent.

**Review by LuLaValva — APPROVED** [review]
Looks good to me

**Review by agliga — APPROVED** [review]

**Pull request merged**